### PR TITLE
NS8 UI

### PR DESCRIPTION
**Pull request by Tbaile**
- Added server-side validation
- Added UI
- Refactor NS8 build process with Containerfile

Screenshots:
image
image

### CODE REVIEW

File: ui/public/metadata.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  "description": {
+    "en": "My kickstart module"
+  },
+  "categories": [],
+  "authors": [
+    {
+      "name": "Name Surname",
+      "email": "author@yourmail.org"
+    }
+  ],
```
Comment (gsanchietti) [inline ui/public/metadata.json:12]: Please enter real data.

File: ui/src/components/AppSideMenu.vue [hunk 2]
```diff
@@ -1,3 +1,7 @@
+<!--
+  Copyright (C) 2022 Nethesis S.r.l.
```
Comment (gsanchietti) [inline ui/src/components/AppSideMenu.vue:2]:
```suggestion
  Copyright (C) 2023 Nethesis S.r.l.
```

File: ui/src/components/AppSideMenuContent.vue [hunk 3]
```diff
@@ -1,3 +1,7 @@
+<!--
+  Copyright (C) 2022 Nethesis S.r.l.
```
Comment (gsanchietti) [inline ui/src/components/AppSideMenuContent.vue:2]:
```suggestion
  Copyright (C) 2023 Nethesis S.r.l.
```

File: ui/src/main.js [hunk 4]
```diff
@@ -1,7 +1,12 @@
+//
+// Copyright (C) 2022 Nethesis S.r.l.
```
Comment (gsanchietti) [inline ui/src/main.js:2]:
```suggestion
// Copyright (C) 2023 Nethesis S.r.l.
```

File: imageroot/actions/configure-module/validate-input.json [hunk 5]
```diff
@@ -3,25 +3,166 @@
...
+        },
         "lets_encrypt": {
-            "type": "boolean",
-            "title": "Let's Encrypt certificate",
-          "description": "Request a valid Let's Encrypt certificate."
+            "description": "Enable Let's Encrypt certificate",
+            "type": "boolean"
+        },
+        "user_domain": {
+            "description": "Domain where the user accessing Nethvoice are stored",
```
Comment (gsanchietti) [inline imageroot/actions/configure-module/validate-input.json:163]:
```suggestion
            "description": "Domain where the user accessing NethVoice are stored",
```

File: imageroot/actions/configure-module/validate-input.json [hunk 6]
```diff
@@ -3,25 +3,166 @@
...
+                    "gateway": {
+                        "description": "Network gateway",
+                        "type": "string",
+                        "format": "ipv4"
+                    }
+                }
+            }
+        },
         "nethvoice_host": {
+            "description": "Host where nethvoice instance should be exposed to",
```
Comment (gsanchietti) [inline imageroot/actions/configure-module/validate-input.json:63]:
```suggestion
            "description": "Host where NethVoice instance should be exposed to",
```

File: imageroot/actions/configure-module/validate-input.json [hunk 7]
```diff
@@ -3,25 +3,166 @@
...
+        }
+    ],
     "type": "object",
     "required": [
         "nethvoice_host",
         "nethcti_ui_host"
     ],
     "properties": {
+        "nethvoice_host_local_networks": {
+            "description": "Local networks where nethvoice instance should be exposed to",
```
Comment (gsanchietti) [inline imageroot/actions/configure-module/validate-input.json:33]:
```suggestion
            "description": "Local networks where NethVoice instance should be exposed to",
```

File: imageroot/actions/configure-module/validate-input.json [hunk 8]
```diff
@@ -3,25 +3,166 @@
     "title": "Configure nethvoice",
     "$id": "http://nethserver.org/json-schema/task/input/nethvoice/configure",
     "description": "Configure nethvoice",
```
Comment (gsanchietti) [inline imageroot/actions/configure-module/validate-input.json:5]:
```suggestion
    "description": "Configure NethVoice",
```

File: imageroot/actions/configure-module/validate-input.json [hunk 9]
```diff
@@ -3,25 +3,166 @@
     "title": "Configure nethvoice",
```
Comment (gsanchietti) [inline imageroot/actions/configure-module/validate-input.json:3]:
```suggestion
    "title": "Configure NethVoice",
```

File: ui/src/router/index.js [hunk 10]
```diff
@@ -1,8 +1,11 @@
+//
+// Copyright (C) 2022 Nethesis S.r.l.
```
Comment (gsanchietti) [inline ui/src/router/index.js:2]:
```suggestion
// Copyright (C) 2023 Nethesis S.r.l.
```

File: ui/src/store/index.js [hunk 11]
```diff
@@ -1,11 +1,15 @@
+//
+// Copyright (C) 2022 Nethesis S.r.l.
```
Comment (gsanchietti) [inline ui/src/store/index.js:2]:
```suggestion
// Copyright (C) 2023 Nethesis S.r.l.
```

File: ui/src/router/index.js [hunk 12]
```diff
@@ -18,11 +21,6 @@ const routes = [
     name: "Settings",
     component: Settings,
   },
-  {
-    path: "/logs",
-    name: "Logs",
-    component: Logs,
-  },
```
Comment (gsanchietti) [inline ui/src/router/index.js:25]: Why Logs view has been removed?

File: ui/src/views/About.vue [hunk 13]
```diff
@@ -1,20 +1,28 @@
+<!--
+  Copyright (C) 2022 Nethesis S.r.l.
```
Comment (gsanchietti) [inline ui/src/views/About.vue:2]:
```suggestion
  Copyright (C) 2023 Nethesis S.r.l.
```

File: ui/src/views/Settings.vue [hunk 14]
```diff
@@ -1,63 +1,103 @@
+<!--
+  Copyright (C) 2022 Nethesis S.r.l.
```
Comment (gsanchietti) [inline ui/src/views/Settings.vue:2]:
```suggestion
  Copyright (C) 2023 Nethesis S.r.l.
```

File: ui/src/views/Settings.vue [hunk 15]
```diff
@@ -1,63 +1,103 @@
...
+              :invalid-message="error.nethcti_ui_host"
+              ref="nethcti_ui_host"
+            />
+            <cv-toggle
+              :label="$t('settings.lets_encrypt')"
+              value="lets_encrypt"
+              :disabled="loading.getConfiguration || loading.configureModule"
+              v-model="form.lets_encrypt"
+            />
+            <!-- TODO: use cluster/list-user-domains to list available domains, otherwise allow manual imput --->
```
Comment (gsanchietti) [inline ui/src/views/Settings.vue:48]: Let's remove the TODO and implement it.

File: ui/src/views/Status.vue [hunk 16]
```diff
@@ -1,59 +1,70 @@
+<!--
+  Copyright (C) 2022 Nethesis S.r.l.
```
Comment (gsanchietti) [inline ui/src/views/Status.vue:2]:
```suggestion
  Copyright (C) 2023 Nethesis S.r.l.
```

**Review by gsanchietti — CHANGES_REQUESTED** [review]
Another question: why favicon and logo have been removed?

File: ui/src/router/index.js (hunk 12 above)
Comment (Tbaile) [reply ui/src/router/index.js:25]: No longer used in modern UI, link to logging interface is used instead

File: ui/src/views/Settings.vue (hunk 15 above)
Comment (Tbaile) [reply ui/src/views/Settings.vue:48]: Will take a bit of time, due to complexity of the component, but I'm on it

**Comment by Tbaile** [comment]
Updated Settings UI:
image

File: ui/src/views/Settings.vue [hunk 17]
```diff
@@ -1,63 +1,142 @@
...
+            />
+            <cv-text-input
+              v-if="form.user_domain_select == 'external'"
+              :label="$t('settings.user_domain')"
+              v-model="form.user_domain"
+              placeholder="domain.example.com"
+              :disabled="loadingState"
+              :invalid-message="error.user_domain"
+              ref="user_domain"
+            />
```
Comment (gsanchietti) [inline ui/src/views/Settings.vue:94]: What is this combox-box for?
A user domain should always be accessed using ldapproxy (local nginx instance).
For all modules, the user domain is always provided by the cluster in clear-text.

If you want to connect to an external provider, you should configured an external user domain.

**Review by gsanchietti — CHANGES_REQUESTED** [review]
Also, please consider that after saving the configuration, the user domain is not shown anymore in the UI.

image

**Comment by Tbaile** [comment]
Allowed selection on only cluster-provided domains

**Review by gsanchietti — APPROVED** [review]

**Review by Stell0 — APPROVED** [review]

**Pull request merged**